### ISSUE TITLE
Discrepancy between EventItems and PDF version of agenda

### ISSUE THREAD

**Issue by reginafcompton** [datamade/la-metro-councilmatic#337]
The July 18 meeting of the Finance, Budget, and Audit Committee shows a discrepancy between the posted agenda and the [EventItems (shown as related board reports in Councilmatic)](http://webapi.legistar.com/v1/metro/events/1366/eventitems): https://boardagendas.metro.net/event/finance-budget-and-audit-committee-e911cf6f4bcb/

Granicus told Metro that this type of error can happen due to duplicate data in their system (i.e., line items in the database can be numbered the same, so the data looks like a duplicate and the system ignores it). They did not provide an explanation for why this can happen. 

Metro now runs a script, which checks for and solves this discrepancy, before publishing an agenda. 

In addition, DataMade may be able to alert Metro of these discrepancies using the following strategies:
1. finding a clue in the API for the abovementioned duplicates - @shrayshray can you tell me specifically which field showed the duplicate? or is it not visible in the Legistar API? (I do not see duplicate `EventItemAgendaSequence` in the API, so I might be overlooking something: http://webapi.legistar.com/v1/metro/events/1366/eventitems)
2. parsing the PDF for event items and comparing those results to the items scrapes from Legistar. DataMade will show the results (if "red") on the [forthcoming integrity dashboard](https://github.com/datamade/la-metro-councilmatic/issues/257).

The second option would require significant time. Let's determine if the first option is possible, first.

**Comment by shrayshray** [comment]
@reginafcompton an item number was duplicated, so EventItemAgendaSequence makes sense as the API field where this would be stored ... is it possible you're not seeing the duplicate now because it has since been corrected in Legistar?

**Comment by reginafcompton** [comment]
That does seem possible - I did not realize that Granicus corrected the problem. Are you be able to contact Granicus and ask specifically about the name of duplicate field (and if it is, indeed, `EventItemAgendaSequence`)?

**Comment by shrayshray** [comment]
@reginafcompton I'm still waiting on a response from Granicus for confirmation on the name of the field, will let you know as soon as I hear back.

**Comment by reginafcompton** [comment]
@shrayshray can you clarify - is it the `EventItemAgendaSequence` that provides the line number? Or am I misreading your comment?

**Comment by shrayshray** [comment]
@reginafcompton sorry! I copied and pasted from Granicus's response and the field name didn't show up. Yes, EventItemAgendaSequence provides the agenda line number.

**Comment by reginafcompton** [comment]
Great - we could implement my first suggestion (see above). 

It seems like the best place to catch these duplicates is at the scraper level, specifically, when we iterate over [the `sorted_items` list](https://github.com/opencivicdata/python-legistar-scraper/blob/master/legistar/events.py#L203). Then, we could (1) raise an error or warning, and email Metro. Or (2) we could add a note in the event extras field, import that note to Councilmatic, and then render a warning on [the forthcoming report dashboard](https://github.com/datamade/la-metro-councilmatic/issues/257). 

The second option would require a little extra work, but would insure a fully automated process.

**Comment by shrayshray** [comment]
@reginafcompton with option 2, could there also be an email alert sent to Metro? With either option, would the raising of an error or a note in the event extras field preclude the flagged item or the event/Agenda from posting on the website ... until it's reviewed/approved, or the like?

**Comment by reginafcompton** [comment]
@shrayshray I implemented solution (1), wherein the scraper fails – since this seems like a significant (albeit rare) data error, and we want to insure that duplicates do not appear on the site. 

In sum, the scrapers look for EventAgendaItems with duplicate Sequence numbers. If a duplicate exists, the scrape fails and [raises an error, instructing DataMade to contact Metro.](https://github.com/datamade/scrapers-us-municipal/blob/master/lametro/events.py#L220) 

I also updated issue #257 with a suggestion about integrating such errors into the "data integrity" report.

I'll close this, but let me know if you have questions @shrayshray !
